indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -134,6 +134,23 @@ default int getPriority()
    */
   String getType();
 
+  /**
+   * Provides a default implementation for obtaining a task's label. This label is intended
+   * to serve as a descriptive identifier for the task's purpose or type, which can be utilized
+   * for more nuanced task management, monitoring, and logging activities.
+   *
+   * By default, this method returns the task type obtained from {@link #getType()},
+   * reflecting the basic categorization of the task.
+   *
+   * @return A string representing the task's label, which by default is the task type.
+   *         Override this method to specify more granular labels tailored to the task's
+   *         specific purpose or operational characteristics.
+   */
+  default String getLabel()
```

Comment:
It seems weird that we are having an implementation in a contrib extension and not in the core overlord task runner. 
cc @kfaraz 
Wdyt about this approach.